deepspeed/runtime/pipe/engine.py

```diff
@@ -65,15 +65,15 @@ class PipelineEngine(DeepSpeedEngine):
     ]
     DTYPE_TO_ID = {dtype: id_ for id_, dtype in enumerate(ID_TO_DTYPE)}
 
-    def __init__(self, has_bool_tensors=False, *super_args, **super_kwargs):
+    def __init__(self, *super_args, **super_kwargs):
         super().__init__(*super_args, **super_kwargs)
         assert isinstance(self.module, PipelineModule), "model must base PipelineModule"
 
         assert self.zero_optimization_stage() < 2, "ZeRO-2 and ZeRO-3 are incompatible with pipeline parallelism"
 
         # We schedule the all-reduces, so disable it in super().backward()
         self.enable_backward_allreduce = False
-        self.has_bool_tensors = has_bool_tensors
+        self.has_bool_tensors = super_kwargs.get("send_bool_tensors", False)
```

Comment:
4. Move `has_bool_tensors` to `kwargs` to make usable with `deepspeed.initialize` and rename `has` to `send`. see https://github.com/microsoft/DeepSpeed/pull/1399